tests/upgrade_test.go

```diff
@@ -0,0 +1,122 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Jan Christoph Uhde <jan@uhdejc.com>
+//
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dchest/uniuri"
+
+	driver "github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	kubeArangoClient "github.com/arangodb/kube-arangodb/pkg/client"
+	"github.com/arangodb/kube-arangodb/pkg/util"
+)
+
+// TODO - environments (provided from outside)
```

Comment:
I don't understand. What is there TODO?

Perhaps you can replace this by a comment, saying that the kubernetes deployment environments are controlled outside the tests and are therefor not a variable in these tests.